java/src/main/native/OrtJniUtil.c

```diff
@@ -534,7 +535,10 @@ size_t copyPrimitiveArrayToJava(JNIEnv *jniEnv, ONNXTensorElementDataType onnxTy
             return consumedSize;
         }
         case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT16: { // stored as a uint16_t
-            float *floatArr = malloc(sizeof(float) * outputLength);
+            jfloat *floatArr = malloc(sizeof(jfloat) * outputLength);
+            if(floatArr == NULL) {
+                throwOrtException(jniEnv, 1, "Not enough memory");
```

Comment:
>throwOrtException [](http://example.com/codeflow?start=16&length=17)

Needs a return statement after the function returns. We just set a flag and do not throw here.